mod-source-record-manager-server/src/main/java/org/folio/services/journal/JournalUtil.java

```diff
@@ -59,49 +64,82 @@ record = new Record()
       } else {
         record = new ObjectMapper().readValue(recordAsString, Record.class);
       }
-      String entityAsString = eventPayloadContext.get(entityType.value());
+      List<JournalRecord> journalRecords = new LinkedList<>();
 
-      JournalRecord journalRecord = new JournalRecord()
-        .withJobExecutionId(record.getSnapshotId())
-        .withSourceId(record.getId())
-        .withSourceRecordOrder(record.getOrder())
-        .withEntityType(entityType)
-        .withActionType(actionType)
-        .withActionDate(new Date())
-        .withActionStatus(actionStatus);
-
-      if (!isEmpty(entityAsString)) {
-        JsonObject entityJson = new JsonObject(entityAsString);
-        journalRecord.setEntityId(entityJson.getString("id"));
-
-        if (entityType == INSTANCE || entityType == HOLDINGS || entityType == ITEM || entityType == PO_LINE) {
-          if (entityType == HOLDINGS) {
-            journalRecord.setInstanceId(entityJson.getString("instanceId"));
-          }
-          if (entityType == ITEM) {
-            if (eventPayloadContext.containsKey(INSTANCE.value())) {
-              JsonObject instanceJson = new JsonObject(eventPayloadContext.get(INSTANCE.value()));
-              journalRecord.setInstanceId(instanceJson.getString("id"));
-            } else if (eventPayloadContext.containsKey(HOLDINGS.value())) {
-              JsonObject holdingsJson = new JsonObject(eventPayloadContext.get(HOLDINGS.value()));
-              journalRecord.setInstanceId(holdingsJson.getString("instanceId"));
-            }
-            journalRecord.setHoldingsId(entityJson.getString("holdingsRecordId"));
-          }
-          if (entityType == PO_LINE){
-            journalRecord.setOrderId(entityJson.getString("purchaseOrderId"));
-          }
-          journalRecord.setEntityHrId(entityJson.getString("hrid"));
+      if(isNotBlank(eventPayloadContext.get("NOT_MATCHED_NUMBER"))) {
+        int notMatchedNumber = Integer.parseInt(eventPayloadContext.get("NOT_MATCHED_NUMBER"));
+        for (int i = 0; i<notMatchedNumber;i++) {
+          journalRecords.add(constructBlankJournalRecord(record, JournalRecord.ActionType.NON_MATCH,entityType,actionStatus)
+            .withEntityType(entityType)
+            .withEntityId(UUID.randomUUID().toString()));
```

Comment:
propose not to set a random uuid as entityId to journalRecord when holdings/items are not found